website/events/api/v2/admin/views.py

```diff
@@ -0,0 +1,145 @@
+import json
+
+from django.http import HttpResponse, Http404
+from oauth2_provider.contrib.rest_framework import IsAuthenticatedOrTokenHasScope
+from rest_framework import status
+from rest_framework.generics import get_object_or_404
+from rest_framework.views import APIView
+from rest_framework import filters as framework_filters
+
+from events import services
+from events.api.v2.admin import filters
+from events.api.v2.admin.permissions import IsOrganiser
+from events.api.v2.admin.serializers.event import EventListSerializer, EventSerializer
+from events.api.v2.admin.serializers.event_registration import (
+    EventRegistrationSerializer,
+)
+from events.models import Event, EventRegistration
+from thaliawebsite.api.v2.admin.views import (
+    AdminListAPIView,
+    AdminRetrieveAPIView,
+    AdminCreateAPIView,
+    AdminUpdateAPIView,
+    AdminDestroyAPIView,
+)
+import events.api.v2.filters as normal_filters
+
+
+class EventAdminListAPIView(AdminListAPIView):
+    queryset = Event.objects.prefetch_related("organiser")
```

Comment:
Still quite a hell lot of queries that are not really necessary

<img width="1411" alt="image" src="https://user-images.githubusercontent.com/7915741/120525372-5c127d80-c3d8-11eb-855c-08e4cca2361e.png">

Can probably be fixed with prefetching the event registrations as well. But we could also fix it by making the num_participants a queryable property which I'm more in favor of (#1617 which should probably be renamed 'queryable properties for active members and events'). So you can also ignore it here if you want

Reply:
I tried adding the `eventregistration_set` to the prefetch but that added a query instead of removing one. Which is understandable since the queries you see here are the registration information field which are completely independent models. Not sure I can prefetch those through the event registrations.

Do agree that queryable properties are probably nice, but really out of scope for this PR.